moto/iam/models.py

```diff
@@ -946,6 +946,14 @@ def _users(self):
         return len(self._iam_backend.users)
 
 
+def filter_items_with_path_prefix(path_prefix, items):
+    filtered_items = []
+    for role in items:
+        if role.path.startswith(path_prefix):
+            filtered_items.append(role)
+    return filtered_items
```

Comment:
I think this is simple enough to convert to a list-comprehension:
`return [role for role in items if role.path.startswith(path_prefix)]`